src/framework/constraints.ml

```diff
@@ -1056,27 +1056,102 @@ module DeadBranchLifter (S: Spec): Spec =
 struct
   include S
 
-  let name () = "DeadBranch (" ^ name () ^ ")"
+  let name () = "DeadBranch (" ^ S.name () ^ ")"
 
-  module Locmap = Deadcode.Locmap
+  module V =
+  struct
+    include Printable.Either (S.V) (Node)
+    let s x = `Left x
+    let node x = `Right x
+  end
+
+  module EM =
+  struct
+    include MapDomain.MapBot (Basetype.CilExp) (Basetype.Bools)
+    let leq x y = !GU.postsolving || leq x y (* HACK: to pass verify*)
+  end
 
-  let dead_branches = function true -> Deadcode.dead_branches_then | false -> Deadcode.dead_branches_else
+  module G =
+  struct
+    include Lattice.Lift2 (S.G) (EM) (Printable.DefaultNames)
+
+    let s = function
+      | `Bot -> S.G.bot ()
+      | `Lifted1 x -> x
+      | _ -> failwith "DeadBranchLifter.s"
+    let node = function
+      | `Bot -> EM.bot ()
+      | `Lifted2 x -> x
+      | _ -> failwith "DeadBranchLifter.node"
+    let create_s s = `Lifted1 s
+    let create_node node = `Lifted2 node
+
+    let printXml f = function
+      | `Lifted1 x -> S.G.printXml f x
+      | `Lifted2 x -> BatPrintf.fprintf f "<analysis name=\"dead-branch\">%a</analysis>" EM.printXml x
+      | x -> BatPrintf.fprintf f "<analysis name=\"dead-branch-lifter\">%a</analysis>" printXml x
```

Comment:
It is on purpose that the analysis names are different here?

Reply:
This shouldn't be needed in proper cases anyway, but I didn't want the top and bot output going under dead-branch, because that would mix them up in the output with the top and bot of the lifted boolean lattice, making it incredibly hard to debug, if anything ever goes wrong with it.